src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/FrameOfT.cs

```diff
@@ -40,26 +47,47 @@ public Frame(IHttpApplication<TContext> application, FrameContext frameContext)
 
                     while (!_requestProcessingStopping)
                     {
-                        var result = await Input.ReadAsync();
-                        var examined = result.Buffer.End;
-                        var consumed = result.Buffer.End;
+                        if (needBuffer)
+                        {
+                            result = await Input.ReadAsync();
+                            buffer = result.Buffer;
+                            needBuffer = false;
+                        }
+
+                        var needAdvance = false;
 
                         try
                         {
-                            ParseRequest(result.Buffer, out consumed, out examined);
+                            ParseRequest(ref buffer, out consumed, out examined);
+
+                            if (buffer.Length == 0)
```

Comment:
Back when we still had `SocketInput`, we had a bug where we weren't advancing to the furthest examined position. This caused the server to spin a lot while waiting for the full request, because the `await` always completed immediately.

Wouldn't the same happen here?